### PR TITLE
Fix missing k8s manifests from release artifact

### PR DESCRIPTION
**Pull request by lmm**
For legacy releases, we'll still refer to `_site/vX.Y/manifests`

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -350,10 +350,15 @@ RELEASE_DIR_K8S_MANIFESTS?=$(RELEASE_DIR)/k8s-manifests
 RELEASE_DIR_IMAGES?=$(RELEASE_DIR)/images
 RELEASE_DIR_BIN?=$(RELEASE_DIR)/bin
 
-# Determine where the manifests live. For older versions we used
-# a different location, but we still need to package them up for patch
-# releases.
+# Determine where the manifests live.
+ifneq ("$(wildcard $(RELEASE_STREAM))","")
```
Comment (ozdanborne) [inline Makefile]: can we just delete this conditional altogether now that manifests always exist in `_site/manifests`?

Comment (lmm) [reply Makefile]: Yeah good point. PTAL @ozdanborne

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit a0c31cb68d0192f42c549a2e18a4f4fa14bb9cde

https://deploy-preview-3235--calico.netlify.com

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 8d9d8cbc9dc0c0d0e2d727772f34d1f041f1b385

https://deploy-preview-3235--calico.netlify.com

**Pull request merged**